src/js/components/RenderPromise.js

```diff
@@ -17,7 +17,7 @@ export default class RenderPromise extends React.PureComponent {
     this.executePromise(this.props.promise);
   }
 
-  componentWillReceiveProps (newProps) {
+  static getDerivedStateFromProps (newProps) {
```

Comment:
This wouldn't work as we are in a static function. I think `componentDidUpdate` is the right place to do this.

Reply:
I will discuss this with you to understand it completely as of now, you suggest what is correct i will change it accordingly

Reply:
you want me to change it to `componentDidUpdate`?
```js
  componentDisUpdate (newProps) {
    this.executePromise(newProps.promise);
  }
```

Reply:
`this.props` will be newProps